python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1090,12 +1109,17 @@ def _subset_graph_backed_asset(
 
         return get_graph_subset(self.node_def, op_selection)
 
-    def subset_for(self, selected_asset_keys: AbstractSet[AssetKey]) -> "AssetsDefinition":
-        """Create a subset of this AssetsDefinition that will only materialize the assets in the
-        selected set.
+    def subset_for(
+        self,
+        selected_asset_keys: AbstractSet[AssetKey],
+        selected_asset_check_keys: Optional[AbstractSet[AssetCheckKey]],
+    ) -> "AssetsDefinition":
+        """Create a subset of this AssetsDefinition that will only materialize the assets and checks
+        in the selected set.
```

Comment:
While outside the scope of this PR, and we need to move forward, this function definitely makes me think we need a unified concept of some sort of "keyed entity" (spec?) in the asset graph so that we don't have to thread another parallel set of arguments around.